### PR TITLE
L4t r32.1 cti

### PR DESCRIPTION
**Pull request by Ecordonnier**
I recommend creating a branch l4t-r32.1-cti for this update, or a branch master alternatively.

### CODE REVIEW

File: conf/distro/include/rudi.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DEPLOY_CONF_NAME = "NVIDIA Rudi TX2"
```
Comment (lfdmn) [inline conf/distro/include/rudi.conf]: rename to rudi-tx2.conf

File: conf/layer.conf [hunk 2]
```diff
@@ -12,4 +12,5 @@ BBFILE_PRIORITY_connecttech = "10"
 
 
 LAYERDEPENDS_BBFILE_PRIORITY_connecttech = "tegra"
+LAYERSERIES_COMPAT_connecttech = "thud"
```
Comment (lfdmn) [inline conf/layer.conf]: meta-tegra:warrior is also based in l4t-32.1. So warrior can be added in the compatibility list

File: conf/machine/rudi.conf [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#@TYPE: Machine
```
Comment (lfdmn) [inline conf/machine/rudi.conf]: rename to rudi-tx2.conf

File: recipes-bsp/tegra-binaries/tegra-flashvars_32.1.0.bbappend [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files/rudi:"
```
Comment (lfdmn) [inline recipes-bsp/tegra-binaries/tegra-flashvars_32.1.0.bbappend]: Should be 

`FILESEXTRAPATHS_prepend := "${THISDIR}/files/:"` 

to accept other MACHINE configurations.

See [https://github.com/madisongh/meta-tegra/issues/177](https://github.com/madisongh/meta-tegra/issues/177) and [https://github.com/madisongh/meta-tegra/wiki/Creating-a-custom-MACHINE](https://github.com/madisongh/meta-tegra/wiki/Creating-a-custom-MACHINE)

**Review by lfdmn — COMMENTED** [review]
Salut Etienne,

Which version of the kernel sources from CTI did you use for merging the kernels? We should name the branch l4t-r32.1-cti-vxxx to keep track on changes from CTI.

Today I pushed the l4t-r32.2-cti-v126 I've been working on. Still WIP. I added the rudy MACHINE files from this PR.

Also madisongh made a new repository for the 4.9 tegra kernel. I forked it.

The kernel patch from this PR is too big to be usable if anyone would like to make fixes. So I invite you to make a PR to [https://github.com/lfdmn/linux-tegra-4.9](https://github.com/lfdmn/linux-tegra-4.9) for the kernel and remove the patch from this PR.

Cheers,
-Damien

**Comment by Ecordonnier** [comment]
Salut Damien,

I used http://connecttech.com/ftp/dropbox/cti-l4t-src-v125.tgz which contains CTI sources for 32.1 (this information is already contained in the patch commit message but you probably didn't see it because github hides the diff of "big" files per default).

Thanks for your review, I agree to most of your comments and I'll try to do that tomorrow, but I am not sure that creating a branch l4t-r32.1-cti-vxxx is ideal. I would rather either directly push to l4t-r32.1 or create a branch l4t-r32.1-cti, if cti decides to release a new version of 32.1 e.g. v142 we can just update the branch l4t-r32.1-cti instead of maintaining several branches for each cti version (l4t-r32.1-cti-v125 and l4t-r32.1-cti-v142).

By the way one other thing is that the other machines.conf files probably also need the change of KERNEL_ARGS in order to have a 115200 bauds kernel log (you probably didn't notice if you are using only ssh), but I don't have the hardware to test this.

merci,
Etienne

**Comment by Ecordonnier** [comment]
Salut Damien, I pushed the changes you requested.

**Comment by Ecordonnier** [comment]
Hi Damien, do you still want to merge the PR?
Cheers
Etienne

**Comment by lfdmn** [comment]
Hi Etienne.

Yes for sure, I've just been very busy. I'll have some questions I try to get back to you this week.

Cheers,
-Damien

**Pull request closed without merging**